aptos-move/framework/aptos-framework/sources/util.move

```diff
@@ -2,9 +2,12 @@
 module aptos_framework::util {
     friend aptos_framework::code;
     friend aptos_framework::gas_schedule;
+    friend aptos_framework::type_map;
 
     /// Native function to deserialize a type T.
-    /// TODO: may want to move it in extra module if needed also in other places inside of the Fx.
-    /// However, should not make this function public outside of the Fx.
-    public(friend) native fun from_bytes<T: copy + drop>(bytes: vector<u8>): T;
+    ///
+    /// Note that this function does not put any constraint on `T`. If code uses this function to
+    /// deserialized a linear value, its their responsibility that the data they deserialize is
+    /// owned.
+    public(friend) native fun from_bytes<T>(bytes: vector<u8>): T;
```

Comment:
could we make it more safe by starting with only deserialize primitive types?  If we see new usecase, we can add them later like txn struct arguments.

Reply:
There is nothing unsafe here. No reason for restrictions. These are also standard patterns, they are basically in C++, Rust, protobuf, ... and there are no restrictions there easier.

However, if you want to use primitive types, the thing currently doesn't work because the restrictions of TypeInfo which is unfortunately only a StructInfo today...

Reply:
Yea friend function limited the access to some core modules and less likely to be used maliciously.   I think it is already much better if we have this deserializer in Move and I can use this to deserialize to primitive when necessary.

Reply:
You don't need to call serialization directly anymore with the newest version of this PR, just use the `Any` type. In fact the token package could not call the friend function because it is in a different package.